registry-schema.adoc

```diff
@@ -0,0 +1,418 @@
+= Registry Schemas
+
+== extraDevfileEntries.yaml
+A file called extraDevfileEntries.yaml can be added under the root of the devfile registry source repository to add samples and stacks from other repositories to the registry. This file contains the location information on where the extra samples and stacks can be found.
+
+=== extraDevfileEntries.yaml schema
+- samples: schema[] - The sample list
+- stacks: schema[] - The stack list
+- name: string - The stack name
+- displayName: string - The display name of devfile
+- description: string - The description of devfile
+- tags: string[] - The tags associated to devfile
+- icon: string - The devfile icon
+- language: string - The project language that is used in the devfile
+- git: *git - The information of remote repositories
+
+=== extraDevfileEntries.yaml sample
+[source,yaml]
+----
+schemaVersion: 1.0.0
+samples:
+  - name: nodejs-basic
+    displayName: Basic NodeJS
+    description: A simple Hello World Node.js application
+    icon: https://github.com/maysunfaisal/node-bulletin-board-2/blob/main/nodejs-icon.png
+    tags: ["NodeJS", "Express"]
+    projectType: nodejs
+    language: nodejs
+    git:
+      remotes:
+        origin: https://github.com/redhat-developer/devfile-sample.git
+  - name: code-with-quarkus
+    displayName: Basic Quarkus
+    description: A simple Hello World Java application using Quarkus
+    icon: .devfile/icon/quarkus.png
+    tags: ["Java", "Quarkus"]
+    projectType: quarkus
+    language: java
+    git:
+      remotes:
+        origin: https://github.com/elsony/devfile-sample-code-with-quarkus.git
+  - name: java-springboot-basic
+    displayName: Basic Spring Boot
+    description: A simple Hello World Java Spring Boot application using Maven
+    icon: .devfile/icon/spring-logo.png
+    tags: ["Java", "Spring"]
+    projectType: springboot
+    language: java
+    git:
+      remotes:
+        origin: https://github.com/elsony/devfile-sample-java-springboot-basic.git
+  - name: python-basic
+    displayName: Basic Python
+    description: A simple Hello World application using Python
+    icon: .devfile/icon/python.png
+    tags: ["Python"]
+    projectType: python
+    language: python
+    git:
+      remotes:
+        origin: https://github.com/elsony/devfile-sample-python-basic.git
+----
+
+== index.json
+A file called index.json contains the metadata of stacks and samples in the registry, user can get the stack or sample generic infomration such as name, language and resources. 
```

Comment:
We should maybe also state that it's an array of devfile stack index entries? Not sure how to phrase that. (Basically an array of the type that you defined down below)

Reply:
I can emphasize that from data structure perspective.